src/FinfoMimeTypeDetector.php

```diff
@@ -90,6 +105,48 @@ private function takeSample(string $contents): string
         return (string) substr($contents, 0, $this->bufferSampleSize);
     }
 
+    /**
+     * Fetches a sample of a resource while maintaining its pointer.
+     *
+     * Non-seekable resources are skipped by default because their pointer can't
+     * be maintained.
+     */
+    private function takeResourceSample($contents): string
+    {
+        if (is_resource($contents) && get_resource_type($contents) === 'stream') {
+            $streamMetaData = stream_get_meta_data($contents);
+            if ($streamMetaData['seekable']) {
+                $streamPosition = ftell($contents);
+            } elseif ($this->ignoreNonSeekableStreams) {
+                return '';
+            }
+
+            // Memory optimization: given a length stream_get_contents()
```

Comment:
where can I validate this?

Reply:
https://www.php.net/manual/en/function.stream-get-contents.php#refsect1-function.stream-get-contents-notes